src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/UpToDate/UpToDateCheckStatePersistence.cs

```diff
@@ -63,64 +63,75 @@ protected override async Task DisposeCoreUnderLockAsync(bool initialized)
             }
         }
 
-        public async Task<(int ItemHash, DateTime ItemsChangedAtUtc)?> RestoreStateAsync(string projectPath, IImmutableDictionary<string, string> configurationDimensions, CancellationToken cancellationToken)
+        public async Task<(int ItemHash, DateTime? ItemsChangedAtUtc)?> RestoreItemStateAsync(string projectPath, IImmutableDictionary<string, string> configurationDimensions, CancellationToken cancellationToken)
         {
             await InitializeAsync(cancellationToken);
 
-            return await ExecuteUnderLockAsync<(int ItemHash, DateTime ItemsChangedAtUtc)?>(
+            return await ExecuteUnderLockAsync<(int ItemHash, DateTime? ItemsChangedAtUtc)?>(
                 async token =>
                 {
                     await InitializeDataAsync(token);
 
                     Assumes.NotNull(_dataByConfiguredProject);
 
-                    if (_dataByConfiguredProject.TryGetValue((projectPath, configurationDimensions), out (int ItemHash, DateTime ItemsChangedAtUtc) storedData))
-                        return storedData;
+                    if (_dataByConfiguredProject.TryGetValue((projectPath, configurationDimensions), out (int ItemHash, DateTime? ItemsChangedAtUtc, DateTime? LastSuccessfulBuildStartedAtUtc) storedData))
+                    {
+                        return (storedData.ItemHash, storedData.ItemsChangedAtUtc);
+                    }
 
                     return null;
                 },
                 cancellationToken);
+        }
 
-            async Task InitializeDataAsync(CancellationToken cancellationToken)
```

Comment:
This local function was promoted to class level. Makes the diff a bit messy here, as another pair of Store/Restore methods were added in this location.